lib/ts/signals.cc

```diff
@@ -46,7 +46,7 @@ signal_check_handler(int signal, signal_handler_t handler)
     sigact = (void *)oact.sa_sigaction;
   }
 
-  if (sigact != handler) {
+  if (sigact != (void *)handler) {
```

Comment:
No C-style casts! `static_cast<void*>(handler)` although we should really rethink using `void*` in this context.